src/Microsoft.Health.Fhir.Core/Features/Validation/NarrativeValidator.cs

```diff
@@ -25,11 +26,25 @@ public NarrativeValidator(INarrativeHtmlSanitizer narrativeHtmlSanitizer)
             _narrativeHtmlSanitizer = narrativeHtmlSanitizer;
         }
 
-        public override IEnumerable<ValidationFailure> Validate(PropertyValidatorContext context)
+        public override string Name => "NarrativeValidator";
+
+        public override bool IsValid(ValidationContext<T> context, TProperty value)
+        {
+            var result = Validate(context);
+            List<ValidationFailure> validationFailures = result as List<ValidationFailure> ?? result.ToList();
+            foreach (var validationFailure in validationFailures)
+            {
+                context.AddFailure(validationFailure);
+            }
+
+            return validationFailures.Count == 0;
+        }
+
+        public IEnumerable<ValidationFailure> Validate(ValidationContext<T> context)
```

Comment:
>public [](http://example.com/codeflow?start=8&length=6)

Does it have to be public?

And I would consider to merge IsValid and Validate into one method.

Reply:
Merged IsValid and Validate into one method!